internal/network/manifests_generator_test.go

```diff
@@ -621,3 +621,117 @@ var _ = Describe("disk encryption manifest", func() {
 		})
 	}
 })
+
+var _ = Describe("nic reaaply manifest", func() {
+	var (
+		ctx                   = context.Background()
+		log                   *logrus.Logger
+		ctrl                  *gomock.Controller
+		manifestsApi          *manifestsapi.MockManifestsAPI
+		manifestsGeneratorApi ManifestsGeneratorAPI
+		db                    *gorm.DB
+		dbName                string
+		clusterId             strfmt.UUID
+		cluster               common.Cluster
+	)
+
+	BeforeEach(func() {
+		log = logrus.New()
+		ctrl = gomock.NewController(GinkgoT())
+		manifestsApi = manifestsapi.NewMockManifestsAPI(ctrl)
+		manifestsGeneratorApi = NewManifestsGenerator(manifestsApi, Config{}, db)
+		db, dbName = common.PrepareTestDB()
+		clusterId = strfmt.UUID(uuid.New().String())
+
+		cluster = common.Cluster{
+			Cluster: models.Cluster{
+				ID: &clusterId,
+			},
+		}
+		Expect(db.Create(&cluster).Error).NotTo(HaveOccurred())
+		manifestsApi.EXPECT().V2CreateClusterManifest(gomock.Any(), gomock.Any()).Times(0)
+	})
+
+	AfterEach(func() {
+		ctrl.Finish()
+		common.DeleteTestDB(db, dbName)
+	})
+
+	Context("CreateClusterManifest", func() {
+		It("added when one of the host installs on an iSCSI drive", func() {
+			manifestsApi.EXPECT().CreateClusterManifestInternal(gomock.Any(), gomock.Any(), false).Times(2).Return(&models.Manifest{
+				FileName: "manifest.yaml",
+				Folder:   models.ManifestFolderOpenshift,
+			}, nil)
+			cluster.Cluster.Hosts = []*models.Host{
+				{
+					Inventory: fmt.Sprintf(`{
+						"disks":[
+							{
+								"id": "install-id",
+								"drive_type": "%s"
+							}
+						]
+						}`, models.DriveTypeISCSI),
+					InstallationDiskID: "install-id",
+				},
+				{
+					Inventory: fmt.Sprintf(`{
+						"disks":[
+							{
+								"id": "install-id",
+								"drive_type": "%s"
+							}
+						]
+						}`, models.DriveTypeSSD),
+					InstallationDiskID: "install-id",
+				},
+			}
+			Expect(manifestsGeneratorApi.AddNicReapply(ctx, log, &cluster)).ShouldNot(HaveOccurred())
+		})
+		It("not added when no hosts installs on an iSCSI drive", func() {
+			cluster.Cluster.Hosts = []*models.Host{
+				{
+					Inventory: fmt.Sprintf(`{
+						"disks":[
+							{
+								"id": "install-id",
+								"drive_type": "%s"
+							}
+						]
+						}`, models.DriveTypeSSD),
+					InstallationDiskID: "install-id",
+				},
+				{
+					Inventory: fmt.Sprintf(`{
+						"disks":[
+							{
+								"id": "install-id",
+								"drive_type": "%s"
+							}
+						]
+						}`, models.DriveTypeSSD),
+					InstallationDiskID: "install-id",
+				},
+			}
+			Expect(manifestsGeneratorApi.AddNicReapply(ctx, log, &cluster)).ShouldNot(HaveOccurred())
```

Comment:
can you just add a vlidatation for the error message

Reply:
this one is one an error message